src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -25,140 +25,146 @@ namespace System.Text
             : this(value, DefaultCapacity)
         {
         }
 
-        // Creates a new string builder from the specified string with the specified 
-        // capacity.  If value is a null String (i.e., if it represents 
-        // String.NullString) then the new string builder will also be null 
-        // (i.e., it will also represent String.NullString).
-        // The maximum number of characters this string may contain is set by capacity.
-        // 
-        public StringBuilder(String value, int capacity)
-            : this(value, 0, ((value != null) ? value.Length : 0), capacity)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StringBuilder"/> class.
+        /// </summary>
+        /// <param name="value">The initial contents of this builder.</param>
+        /// <param name="capacity">The initial capacity of this builder.</param>
+        public StringBuilder(string value, int capacity)
+            : this(value, 0, value?.Length ?? 0, capacity)
         {
         }
 
-        // Creates a new string builder from the specifed substring with the specified
-        // capacity.  The maximum number of characters is set by capacity.
-        // 
-        public StringBuilder(String value, int startIndex, int length, int capacity)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StringBuilder"/> class.
+        /// </summary>
+        /// <param name="value">The initial contents of this builder.</param>
+        /// <param name="startIndex">The index to start in <paramref name="value"/>.</param>
+        /// <param name="length">The number of characters to read in <paramref name="value"/>.</param>
+        /// <param name="capacity">The initial capacity of this builder.</param>
+        public StringBuilder(string value, int startIndex, int length, int capacity)
         {
             if (capacity < 0)
             {
-                throw new ArgumentOutOfRangeException(nameof(capacity),
-                    SR.Format(SR.ArgumentOutOfRange_MustBePositive, nameof(capacity)));
+                throw new ArgumentOutOfRangeException(nameof(capacity), SR.Format(SR.ArgumentOutOfRange_MustBePositive, nameof(capacity)));
             }
             if (length < 0)
             {
-                throw new ArgumentOutOfRangeException(nameof(length),
-                    SR.Format(SR.ArgumentOutOfRange_MustBeNonNegNum, nameof(length)));
+                throw new ArgumentOutOfRangeException(nameof(length), SR.Format(SR.ArgumentOutOfRange_MustBeNonNegNum, nameof(length)));
             }
             if (startIndex < 0)
             {
                 throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_StartIndex);
             }
             Contract.EndContractBlock();
 
-            if (value == null)
-            {
-                value = String.Empty;
-            }
+            value = value ?? string.Empty;
```

Comment:
I would prefer we not make changes like this that add additional unnecessary writes.